File: service/shobuServiceLambda/src/shobu/data/sub/Move.java
package shobu.data.sub;

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBDocument;

@DynamoDBDocument
public class Move {
    @DynamoDBAttribute(attributeName = "fullMove")
    public String getFullMove(){ return fullMove; }
    public void setFullMove( String fullMove ){ this.fullMove = fullMove; }
    private String fullMove;
    @DynamoDBAttribute(attributeName = "time")
    public long getTime(){ return time; }
    public void setTime( long time ){ this.time = time; }
    private long time;
}

File: service/pyShobuService/src/data/move.py

import util.time as time

_fullMove = "m"
_time = "t"

def new( fullMove ):
    return {
        _time: time.getNowMs(),
        _fullMove: fullMove
    }

def getFullMove( this ):
    return this[_fullMove]

def getTime( this ) -> int:
    return this[_time]